bowling-game/tabcorp/31-05-2017/test.js

```diff
@@ -0,0 +1,53 @@
+const assert = require('assert');
+
+describe('Bowling Game', () => {
+
+  const createGame = () => {
+    let rollsResult = []
+    return {
+      roll: function(pinsTakenDown) {
+        rollsResult.push(pinsTakenDown)
+      },
+      score: function() {
+        const pinsTakenDownToRolls = function(pinsTakenDown) {
+          const roll = {
+            pinsTakenDown: pinsTakenDown,
+            strike: false
+          }
+          if (pinsTakenDown === 10) {
+            roll.strike = true
+          }
+          return roll;
+        }
+        const toSumOfPinsTakenDown = (previousSum, currentRoll) => {
+          return previousSum + currentRoll.pinsTakenDown;
+        }
+
+        const rolls = rollsResult.map(pinsTakenDownToRolls)
+        return rolls.reduce(toSumOfPinsTakenDown, 0);
```

Comment:
At this point we just stopped doing TDD and started brainstorming how it would be if we did the calculation in the `score()` function using some sort of CQRS approach where `.roll()` is the command that stores the `pinsTakenDown` internally and `.score()` is the query to build a pseudo-projection at runtime.

As you can see, the `pinsTakenDownToRolls` function has no tests at all.